packages/engine.test/src/testView.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Engine } from "@c11/engine.react";
+import { generateImage } from 'jsdom-screenshot';
+
+jest.useFakeTimers();
+
+export async function testView(name: string, { state, View, props }) {
+    test(name, async () => {
+        const root = document.createElement("div")
+        const title = document.createElement("h1")
+        title.innerHTML = name
+        
+        document.body.innerHTML = ""
+        document.body.appendChild(title)
+        document.body.appendChild(root)
+      
+        const engine = new Engine({
+          state: {
+              initial: state
+          },
+          view: {
+            element: <View
+              {...props}
+            />,
+            root
+          },
+        });
+      
+        jest.runAllTimers();
+      
+        const screenshot = await generateImage();
```

Comment:
1. Where are the images stored? Shouldn't the developer pass some config regarding the path?
2. Should these be added to the repo as a commit for regression testing? What is the recommendation here?
3. Can we just save some kind of checksum that would allow partial comparison? e.g. 90% similar?

Reply:
all of those are covered by a `jest-screenshot.json` in the root project for each app that we are developing, you can also set it up in `package.json` but i hate using package.json as a config file

Reply:
ref https://www.npmjs.com/package/jest-screenshot#configuring

Reply:
the configs will eventually be part of the boilerplate for the engine app

Reply:
but why would we do it like that instead of having the config in the library and allowing the developer to extend it as they please?

Reply:
It's not the project's responsibility to provide the configuration for our testing library, that would mean it leaks implementation details.

In the eventuality that we change/remove/update the `jest-screenshot` package the projects will no longer be able to use their system without updating the configuration.

We should either expose a minimal set of options for the developer (e.g. filepath, format, etc) and the rest of the configuration defined internally with the best configuration for testing Engine views.

Reply:
let's be realistic, this is not the best place to focus the time right now